### PR TITLE
Implement error handling for access token

### PR DESCRIPTION
**Pull request by Zain-Abbas1**
__Short description outlining key changes/additions__
- Return empty OAuthToken object is response is not of 2xx
- Add relevant unit tests
- Return null if token is null or empty

__JIRA Ticket Number__
- FA-770

### Type of change

* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

File: src/main/java/uk/gov/ch/oauth/Oauth2.java [hunk 1]
```diff
@@ -30,27 +30,28 @@
...
     private static final String SIGN_IN_INFO = SessionKeys.SIGN_IN_INFO.getKey();
 
-
-
     @Autowired
     public Oauth2(final IIdentityProvider identityProvider, final SessionFactory sessionFactory) {
         this.identityProvider = identityProvider;
         this.sessionFactory = sessionFactory;
         oAuth2StateHandler = new OAuth2StateHandler(this.identityProvider);
+        logger = LoggerFactory.getLogger("docs.developer.ch.gov.uk");
```
Comment (CH-JonathanBlake) [inline src/main/java/uk/gov/ch/oauth/Oauth2.java:46]: You'll remove this before committing to Oauth Library?

File: src/main/java/uk/gov/ch/oauth/Oauth2.java [hunk 2]
```diff
@@ -205,10 +214,27 @@ OAuthToken requestOAuthToken(String code) {
                 .contentType(MediaType.APPLICATION_FORM_URLENCODED)
                 .accept(MediaType.APPLICATION_JSON)
                 .body(bodyInsert)
-                .retrieve()
-                .bodyToMono(OAuthToken.class);
+                .exchange()
+                .flatMap(this::validateTokenResponse);
```
Comment (CH-JonathanBlake) [inline src/main/java/uk/gov/ch/oauth/Oauth2.java:218]: Could you explain this to me?

File: src/main/java/uk/gov/ch/oauth/Oauth2.java (hunk 1 above)
Comment (Zain-Abbas1) [reply src/main/java/uk/gov/ch/oauth/Oauth2.java:46]: Yeah or rename the var to oauth... if the OAuthCoordinator isn't in yet.

File: src/main/java/uk/gov/ch/oauth/Oauth2.java (hunk 2 above)
Comment (CH-JonathanBlake) [reply src/main/java/uk/gov/ch/oauth/Oauth2.java:218]: Explained offline. Thanks Zain.

File: src/main/java/uk/gov/ch/oauth/Oauth2.java (hunk 1 above)
Comment (CH-JonathanBlake) [reply src/main/java/uk/gov/ch/oauth/Oauth2.java:46]: Might want to speak with Rob about what his changes for Coordinator will look like.

**Pull request merged**